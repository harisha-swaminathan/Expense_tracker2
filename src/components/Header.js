import React from 'react';
import { NavLink } from 'react-router-dom';

const navLinkStyleSelected = ({isActive}) => {
    if (isActive) {
        return `is-active`
    }
};

const Header = () => (
    <header>
    <h1> Hello! I'm the Header. </h1>
    <NavLink to="/" className={navLinkStyleSelected}> Go home!</NavLink> <br/>
    <NavLink to="/create" className={navLinkStyleSelected}> Add an expense</NavLink> <br/>
    <NavLink to="/help" className={navLinkStyleSelected}> Get help</NavLink> <br/>
    </header>
);

export default Header;