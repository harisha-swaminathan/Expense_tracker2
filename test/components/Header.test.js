import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Header from '../../src/components/Header';

test('component is rendered as expected', () =>{
    const { asFragment } = render(<BrowserRouter> <Header /> </BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
});