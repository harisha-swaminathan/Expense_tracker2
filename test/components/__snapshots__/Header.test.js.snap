// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component is rendered as expected 1`] = `
<DocumentFragment>
   
  <header>
    <h1>
       Hello! I'm the Header. 
    </h1>
    <a
      aria-current="page"
      class="is-active"
      href="/"
    >
       Go home!
    </a>
     
    <br />
    <a
      href="/create"
    >
       Add an expense
    </a>
     
    <br />
    <a
      href="/help"
    >
       Get help
    </a>
     
    <br />
  </header>
   
</DocumentFragment>
`;
